### DISCUSSION TITLE
Provide a default versioning (template) for custom datasource

### DISCUSSION THREAD

**Discussion by astellingwerf** [renovatebot/renovate#23949] [Suggest an Idea]
### Tell us more.

For a custom datasource that I'm adding to my corporate Renovate configuration, the versioning scheme for all artifact could (if not should) be shared for all consumers. They would need to add a `regexManager` to find the dependency into their local configuration, because the location and format where the version number is found is unique per instance, but I'd dislike it if they would all need to repeat the shared versioning template for it.

I think that that situation is not unique to mine, and would be beneficial to a larger Renovate audience. I'd say my corporate level configuration would look like this:
```
{
  "customDatasources": {
    "ghs" : {
      "defaultRegistryUrlTemplate": "https://server/artifactory/repo/renovate/astellin/ghs/{{ replace ':' '/' packageName }}.json",
      "defaultVersioningTemplate": "regex:^(?<major>\\d+)0(?<minor>\\d+)0(?<patch>\\d+)$"
    }
  },
  "packageRules": [
    {
      "matchDatasources": ["custom.ghs"],
      "commitMessageTopic": "📘 help reference {{packageName}}"
    }
  ]
}
```
...so that the repo configuration would be just:
```
{
  "regexManagers": [
    {
      "fileMatch": ["(^|/)values.yaml$"],
      "matchStrings": [
        ":\\s+\"(?<currentValue>.*)\"\\s+#\\s+renovate\\s+datasource=(?<datasource>.*?)\\s+depName=(?<depName>.*?)\\s"
      ]
    }
  ]
}
```

**Comment by secustor** [comment]
Sounds good to me.

**Reply by secustor** [reply]
https://github.com/renovatebot/renovate/issues/23950

**Comment by viceice** [comment]
you can use the package rule to assign the versioning inside the present. you already have a rule for commit message, so you just need to add the versioning there

**Reply by astellingwerf** [reply]
In that case, you can probably judge better whether we should proceed with this then. I noticed there are plans to make regex "just one of the custom managers"?

**Reply by secustor** [reply]
It still makes sense as setting a `defaultVersioningTemplate` would work the same as the versioning set in code of other datasources.

When we introduce other custom datasources you can use this additional matcher `matchManagers: ["custom.*"]`
